diff -rupN jack-1.9.8.old/common/JackControlAPI.cpp jack-1.9.8/common/JackControlAPI.cpp
--- jack-1.9.8.old/common/JackControlAPI.cpp	2011-12-19 06:54:02.000000000 -0500
+++ jack-1.9.8/common/JackControlAPI.cpp	2012-04-07 15:04:06.882823048 -0400
@@ -678,7 +678,7 @@ SERVER_EXPORT jackctl_server_t * jackctl
         goto fail_free_parameters;
     }
 
-    value.i = 10;
+    value.i = 60;
     if (jackctl_add_parameter(
             &server_ptr->parameters,
             "realtime-priority",
diff -rupN jack-1.9.8.old/common/JackServerGlobals.cpp jack-1.9.8/common/JackServerGlobals.cpp
--- jack-1.9.8.old/common/JackServerGlobals.cpp	2011-12-19 06:54:02.000000000 -0500
+++ jack-1.9.8/common/JackServerGlobals.cpp	2012-04-07 15:05:11.841498738 -0400
@@ -98,7 +98,7 @@ bool JackServerGlobals::Init()
 {
     int realtime = 0;
     int client_timeout = 0; /* msecs; if zero, use period size. */
-    int realtime_priority = 10;
+    int realtime_priority = 60;
     int verbose_aux = 0;
     int do_mlock = 1;
     unsigned int port_max = 128;
