--- config/cpu/powerpc/cycles.h.orig	2006-05-26 04:45:16.000000000 +0200
+++ config/cpu/powerpc/cycles.h	2008-03-08 18:15:19.000000000 +0100
@@ -36,16 +36,8 @@
 {
 	cycles_t ret = 0;
 
-	__asm__ __volatile__(
-		"98:	mftb %0\n"
-		"99:\n"
-		".section __ftr_fixup,\"a\"\n"
-		"	.long %1\n"
-		"	.long 0\n"
-		"	.long 98b\n"
-		"	.long 99b\n"
-		".previous"
-		: "=r" (ret) : "i" (CPU_FTR_601));
+	__asm__ __volatile__("mftbl %0" : "=r" (ret));
+
 	return ret;
 }
 
