diff -Nur jack-1.9.6-orig/common/JackNetOneDriver.cpp jack-1.9.6/common/JackNetOneDriver.cpp
--- jack-1.9.6-orig/common/JackNetOneDriver.cpp	2010-08-30 12:23:53.000000000 +0200
+++ jack-1.9.6/common/JackNetOneDriver.cpp	2011-02-19 23:55:14.000000000 +0100
@@ -161,7 +161,7 @@
 #if HAVE_CELT_API_0_7
 	    celt_int32 lookahead;
 	    CELTMode *celt_mode = celt_mode_create( netj.sample_rate, netj.period_size, NULL );
-	    netj.capture_srcs = jack_slist_append(netj.capture_srcs, celt_decoder_create( celt_mode, 1, NULL ) );
+	    netj.capture_srcs = jack_slist_append(netj.capture_srcs, celt_decoder_create_custom( celt_mode, 1, NULL ) );
 #else
 	    celt_int32_t lookahead;
 	    CELTMode *celt_mode = celt_mode_create( netj.sample_rate, 1, netj.period_size, NULL );
@@ -209,7 +209,7 @@
 #if HAVE_CELT
 #if HAVE_CELT_API_0_7
 	    CELTMode *celt_mode = celt_mode_create( netj.sample_rate, netj.period_size, NULL );
-	    netj.playback_srcs = jack_slist_append(netj.playback_srcs, celt_encoder_create( celt_mode, 1, NULL ) );
+	    netj.playback_srcs = jack_slist_append(netj.playback_srcs, celt_encoder_create_custom( celt_mode, 1, NULL ) );
 #else
 	    CELTMode *celt_mode = celt_mode_create( netj.sample_rate, 1, netj.period_size, NULL );
 	    netj.playback_srcs = jack_slist_append(netj.playback_srcs, celt_encoder_create( celt_mode ) );
